ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-bgp.j2

```diff
@@ -777,6 +777,15 @@ router bgp {{ router_bgp.as }}
 {%         endif %}
 {%         if vrf.evpn_multicast is arista.avd.defined(true) %}
       evpn multicast
+{%             if vrf.evpn_multicast_address_family.ipv4 is arista.avd.defined %}
+         address-family ipv4
+{%                 if vrf.evpn_multicast_address_family.ipv4.disabled is arista.avd.defined(true) %}
+            disabled
+{%                 endif %}
+{%                 if vrf.evpn_multicast_address_family.ipv4.transit is arista.avd.defined(true) %}
+            transit
+{%                 endif %}
+{%             endif %}
```

Comment:
If both `disabled` or `transit` are `false` but ipv4 is defined the rendered config will be:

```
   vrf blah
      evpn multicast
        address-family ipv4
```

but in my vEOS the actual config is:

```
   vrf blah
      evpn multicast
```

Reply:
Looking at the behavior of disabled on my vEOS instance:

```
DC1-LEAF1A(config)#router bgp 65000.1
DC1-LEAF1A(config-router-bgp)#vrf bloh
! IP routing not enabled
DC1-LEAF1A(config-router-bgp-vrf-bloh)#evpn multicast
DC1-LEAF1A(config-evpn-mcast-vrf-bloh)#show active
router bgp 65000.1
   vrf bloh
      evpn multicast
DC1-LEAF1A(config-evpn-mcast-vrf-bloh)#address-family ipv4
DC1-LEAF1A(config-evpn-mcast-vrf-bloh-af)#show active
DC1-LEAF1A(config-evpn-mcast-vrf-bloh-af)#disabled
DC1-LEAF1A(config-evpn-mcast-vrf-bloh-af)#show active
DC1-LEAF1A(config-evpn-mcast-vrf-bloh-af)#exit
DC1-LEAF1A(config-evpn-mcast-vrf-bloh)#show active
DC1-LEAF1A(config-evpn-mcast-vrf-bloh)#exit
DC1-LEAF1A(config-router-bgp-vrf-bloh)#vrf bloh
! IP routing not enabled
DC1-LEAF1A(config-router-bgp-vrf-bloh)#show active
router bgp 65000.1
   vrf bloh
```

so looks like when only disabled is true then the full block is gone (sounds like a silly config I know.. :) )


Here is what I can see rendered (and it. may not be the same on all platforms:)

```
# transit set and disabled set
router bgp 65000.1
   vrf bloh
      evpn multicast
         address-family ipv4
            transit
            disabled
```

```
# transit set and disabled unset
router bgp 65000.1
   vrf bloh
      evpn multicast
         address-family ipv4
            transit
```

```
# transit unset and disable set
router bgp 65000.1
   vrf bloh
```

```
# none set 
router bgp 65000.1
   vrf bloh
      evpn multicast
```

Not sure to be honest if we want to handle this logic or if what we have currently give enough liberty to a use who wants to configure something that makes sense and not enabling a function to disable it later.

Reply:
> 

Good point about them both being disabled - just to get rid of the empty address-family. Will fix that.

Reply:
The disabled knob will only make sense when we also have ipv6, so remove it for now. We have the global enable.

Reply:
then you can simplify your check for transit in a single `if`

Reply:
@jonxstill I suggest that we remove everything around the `disabled` keyword, since it has no special function compared to just _not_ enable `evpn_multicast: true` in the first place. And it will actually _remove_ config which is not very intuitive (as you noted).

Reply:
OK, will do.

Reply:
Just removed the disabled option.